### PR TITLE
Use meson build system instead of Makefile

### PR DESCRIPTION
**Pull request by linuxdasein**

### CODE REVIEW

**Comment by linuxdasein** [comment]
I've already merged it, yet for some unfathomable reason it stayed open.

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  'src/input.cpp',
+  'src/jsoncpp.cpp',
+  'src/main.cpp',
+  'src/mania.cpp',
+  'src/osu.cpp',
+  'src/system.cpp',
+  'src/taiko.cpp',
+])
+
+cpp_flags = ['-std=c++17']
```
Comment (eli-schwartz) [inline meson.build:19]: You probably want to set this via the project options:

```meson
project(
    ...,
    default_options: ['cpp_std=c++17'],
)
```

File: .gitignore [hunk 2]
```diff
@@ -2,7 +2,7 @@
 *.exe
 *.o
 bongo
-bin/*
+build/*
```
Comment (eli-schwartz) [inline .gitignore:5]: https://mesonbuild.com/FAQ.html#how-do-i-ignore-the-build-directory-in-my-vcs

:) Not necessary for modern meson versions.

File: meson.build (hunk 1 above)
Comment (linuxdasein) [reply meson.build:19]: Thanks for the review, I will take that into account next time I'll be editing meson files.

**Pull request closed without merging**